tests/Resources/ti.ui.window.test.js

```diff
@@ -0,0 +1,500 @@
+
+				finish();
+			} catch (err) {
+				finish(err);
+			}
+		});
+		win.open();
+	});
+
+	// Times out on Android
+	it.androidBroken('window_navigation', function (finish) {
+		var rootWindowFocus = 0;
+		var rootWindowBlur = 0;
+		var rootWindowOpen = 0;
+		var rootWindowClose = 0;
+		var secondWindowFocus = 0;
+		var secondWindowBlur = 0;
+		var secondWindowOpen = 0;
+		var secondWindowClose = 0;
+		var thridWindowFocus = 0;
+		var thridWindowBlur = 0;
+		var thridWindowOpen = 0;
+		var thridWindowClose = 0;
+
+		var rootWindow = Ti.UI.createWindow({
+			backgroundColor: 'navy'
+		});
+
+		rootWindow.addEventListener('focus', function () { rootWindowFocus++; });
+		rootWindow.addEventListener('blur', function () { rootWindowBlur++; });
+		rootWindow.addEventListener('open', function () { rootWindowOpen++; });
+		rootWindow.addEventListener('close', function () { rootWindowClose++; });
+		rootWindow.open();
+
+		setTimeout(function () {
+			var secondWindow = Ti.UI.createWindow({
+				backgroundColor: 'pink'
+			});
+			secondWindow.addEventListener('focus', function () { secondWindowFocus++; });
+			secondWindow.addEventListener('blur', function () { secondWindowBlur++; });
+			secondWindow.addEventListener('open', function () { secondWindowOpen++; });
+			secondWindow.addEventListener('close', function () { secondWindowClose++; });
+			secondWindow.open();
+
+			setTimeout(function () {
+				var thirdWindow = Ti.UI.createWindow({
+					backgroundColor: 'green'
+				});
+				thirdWindow.addEventListener('focus', function () { thridWindowFocus++; });
+				thirdWindow.addEventListener('blur', function () { thridWindowBlur++; });
+				thirdWindow.addEventListener('open', function () { thridWindowOpen++; });
+				thirdWindow.addEventListener('close', function () { thridWindowClose++; });
+				thirdWindow.open();
+
+				setTimeout(function () {
+					thirdWindow.close();
+					setTimeout(function () {
+						secondWindow.close();
+						setTimeout(function () {
+							rootWindow.close();
+							try {
+								should(rootWindowFocus).be.eql(2);
+								should(rootWindowBlur).be.eql(2);
+								should(rootWindowOpen).be.eql(1);
+								should(rootWindowClose).be.eql(1);
+
+								should(secondWindowFocus).be.eql(2);
+								should(secondWindowBlur).be.eql(2);
+								should(secondWindowOpen).be.eql(1);
+								should(secondWindowClose).be.eql(1);
+
+								should(thridWindowFocus).be.eql(1);
+								should(thridWindowBlur).be.eql(1);
+								should(thridWindowOpen).be.eql(1);
+								should(thridWindowClose).be.eql(1);
+								finish();
+							} catch (err) {
+								finish(err);
+							}
+						}, 500);
+					}, 500);
+				}, 500);
+			}, 500);
+		}, 500);
+	});
+
+	it('#applyProperties(Object)', function () {
+		win = Ti.UI.createWindow();
+		win.open();
+		should(win.custom).not.exist;
+		win.applyProperties({ custom: 1234 });
+		should(win.custom).be.eql(1234);
+	});
+	
+	it.ios('largeTitleEnabled', function () {
```

Comment:
FYI To make things easier for merging in new tests, you can now just add a `tests/Resources/ti.ui.window.addontest.js` with just the new tests and it'll get auto-picked up (and I don't have to wade through what changed between this overriding file and the original.).

Reply:
This is awesome news Chris, thanks! 🙂